database_cleanup/models/purge_columns.py

```diff
@@ -32,13 +32,23 @@ def purge(self):
             model_pool = self.env[line.model_id.model]
             # Check whether the column actually still exists.
             # Inheritance such as stock.picking.in from stock.picking
-            # can lead to double attempts at removal
+            # can lead to double attempts at removal.
+            # Ignore the column if it is inherited.
             self.env.cr.execute(
-                'SELECT count(attname) FROM pg_attribute '
-                'WHERE attrelid = '
-                '( SELECT oid FROM pg_class WHERE relname = %s ) '
-                'AND attname = %s',
-                (model_pool._table, line.name))
+                """
+                SELECT COUNT(pa.attname)
+                FROM pg_attribute AS pa
+                JOIN pg_class AS pc ON pa.attrelid = pc.oid
+                LEFT JOIN pg_inherits AS pi ON pc.oid = pi.inhrelid
+                WHERE pc.relname = %(table_name)s
+                AND pa.attname = %(column_name)s
+                AND pi.inhrelid IS NULL;""",
```

Comment:
I don't understand this test. It seems to give the same result on both the inherited column 'binding_model_id' as the native column 'link_field_id'. Am I missing something?

```
12base=>      SELECT count(pa.attname)
                FROM pg_attribute AS pa
                JOIN pg_class AS pc ON pa.attrelid = pc.oid
                LEFT JOIN pg_inherits AS pi ON pc.oid = pi.inhrelid
                WHERE pc.relname = 'ir_act_server'
                AND pa.attname = 'binding_model_id'
                AND pi.inhrelid IS NOT NULL;
 count 
-------
     1
(1 row)

12base=>      SELECT count(pa.attname)
                FROM pg_attribute AS pa
                JOIN pg_class AS pc ON pa.attrelid = pc.oid
                LEFT JOIN pg_inherits AS pi ON pc.oid = pi.inhrelid
                WHERE pc.relname = 'ir_act_server'
                AND pa.attname = 'link_field_id'
                AND pi.inhrelid IS NOT NULL;
 count 
-------
     1
(1 row)
```

Reply:
Here is a query to select inherited columns. Note that this is on a migrated database featuring the old `usage` column.
```
SELECT a.attname from pg_attribute a
join pg_inherits i on i.inhrelid = a.attrelid
join pg_attribute as a2
on i.inhrelid = a.attrelid and i.inhparent = a2.attrelid and a.attname = a2.attname
where a.attrelid = 'ir_act_server'::regclass and a.attnum > 0;
            attname            
-------------------------------
 binding_model_id
 binding_type
 create_date
 create_uid
 help
 id
 name
 openupgrade_legacy_11_0_usage
 openupgrade_legacy_9_0_help
 type
 write_date
 write_uid
(12 rows)
```